src/evox/algorithms/so/de_variants/ode.py

```diff
@@ -107,15 +107,19 @@ def ask(self, state):
         tile_R = jnp.tile(R[:, jnp.newaxis], (1, self.dim))
         masks = jnp.where(tile_arange == tile_R, True, masks_init)
 
-        trial_vectors = vmap(
-            partial(
-                self._ask_one, population=state.population, best_index=state.best_index
+        def generate_trial_vector(index, R, random_choices, mask):
+            # 使用 lambda 函数直接传递所有参数
+            return jax.lax.cond(
+                state.counter % 2 == 0,
+                lambda: self._ask_one_even(index, R, state.population, state.best_index, random_choices, mask),
+                lambda: self._ask_one_odd(index, R, state.population, state.best_index, random_choices, mask)
             )
-        )(indices, R, random_choiced=random_choices, mask=masks)
+
+        trial_vectors = vmap(generate_trial_vector)(indices, R, random_choices, masks)
```

Comment:
In this case, don't use cond inside vmap, instead do this:
```python
trial_vectors = jax.lax.cond(
    state.counter % 2 == 0,
    vmap(self._ask_one_even, in_axes=[0, 0, None, None, 0, 0]),
    vmap(self._ask_one_odd, in_axes=[0, 0, None, None, 0, 0]),
    index, R, state.population, state.best_index, random_choices, mask
)
```
Using cond inside vmap is bad because the `state.counter % 2 == 0` is always the same inside vmap, resulting in unnecessary branching overhead.